### PR TITLE
COL-755, profile page gets 'My Assets'; 'Return to Dashboard' from asset view

### PR DESCRIPTION
**Pull request by johncrossman**
https://jira.ets.berkeley.edu/jira/browse/COL-755
UX: https://drive.google.com/file/d/0B9vFhXfb6IlZYVc1aENROVdrc3c/view

* For now, 'Most Impactful' = assets sorted by likes.
* If user clicks to view asset from profile then s/he'll get "Return to Dashboard" on asset view.
* Styling on profile page is a work in progress.

### CODE REVIEW

File: public/app/app.bootstrap.js [hunk 1]
```diff
@@ -60,9 +60,16 @@
...
 
-    // Extract '_id' from embedded toolUrl to support deep-linking from other SuiteC tool or email, possible.
+    // Extract '_id' from embedded toolUrl to support linking from one SuiteC tool to another.
     var m = queryArgs.match(/.*[\?&]_id=([0-9a-zA-Z]+).*/);
     parameters.deep_link_id = (m && m.length > 0) ? m[1] : null;
+    // Optional arg '_referring_tool' indicates origin of link
+    var tool = queryArgs.match(/.*[\?&]_referring_tool=([a-z]+).*/);
+    parameters.referring_tool = (tool && tool.length > 0) ? tool[1] : null;
+    // Optional arg '_referring_id' describes state of the referring tool when exited.
+    var id = queryArgs.match(/.*[\?&]_referring_id=([[0-9a-zA-Z]+).*/);
```
Comment (paulkerschen) [inline public/app/app.bootstrap.js]: @johncrossman Extra square bracket before `0-9`

File: public/app/dashboard/splash.html [hunk 2]
```diff
@@ -2,15 +2,50 @@
...
 
 <div data-ng-if="userAssetActivity">
-  <h3 data-ng-bind="user.id === me.id ? 'My Classmates' : 'Classmates'"></h3>
+  <h3 data-ng-bind="isMyProfile ? 'My Classmates' : 'Classmates'"></h3>
   <div class="activity-timeline"
        data-activity-timeline="userAssetActivity"
        data-labels-width="200"></div>
 </div>
+
+<div class="featured-assets-container" data-ng-if="featuredAssets && featuredAssets.length > 0">
```
Comment (paulkerschen) [inline public/app/dashboard/splash.html]: @johncrossman In a template I think you're safe with just `data-ng-if="featuredAssets.length"`.

File: public/app/shared/toolHrefDirective.js [hunk 3]
```diff
@@ -64,6 +66,13 @@
           }
         };
         var queryArgs = scope.id ? '?_id=' + scope.id : '';
+        if (attrs.referringTool) {
+          queryArgs = !!queryArgs ? queryArgs + '&' : '?';
```
Comment (paulkerschen) [inline public/app/shared/toolHrefDirective.js]: @johncrossman You probably don't need `!!` since `?` will cast to Boolean.

**Comment by johncrossman** [comment]
I combed out the nits, @paulkerschen  Thanks for review.

**Pull request merged**